app/vmui/packages/vmui/src/pages/ExploreLogs/ExploreLogsBody/views/LiveTailingView/LiveTailingView.tsx

```diff
@@ -0,0 +1,226 @@
+import Switch from "../../../../../components/Main/Switch/Switch";
+import GroupLogsItem from "../../../GroupLogs/GroupLogsItem";
+
+const SCROLL_THRESHOLD = 100;
+const scrollToBottom = () => window.scrollTo({
+  top: document.documentElement.scrollHeight,
+  behavior: "instant"
+});
+const throttledScrollToBottom = throttle(scrollToBottom, 200);
+
+const LiveTailingView: FC<ViewProps> = ({ settingsRef }) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const settingButtonRef = useRef<HTMLDivElement>(null);
+
+  const [isAtBottom, setIsAtBottom] = useState(true);
+  const { value: isSettingsOpen, setFalse: closeSettings, setTrue: openSettings } = useBoolean(false);
+  const [searchParams] = useSearchParams();
+  const { setSearchParamsFromKeys } = useSearchParamsFromObject();
+  const [rowsPerPage, setRowsPerPage] = useStateSearchParams(100, "rows_per_page");
+  const [query, _setQuery] = useStateSearchParams("*", "query");
+  const [isCompactTailingStr] = useStateSearchParams(0, "compact_tailing");
+  const isCompactTailingNumber = Boolean(Number(isCompactTailingStr));
+  const {
+    logs,
+    isPaused,
+    error,
+    startLiveTailing,
+    stopLiveTailing,
+    pauseLiveTailing,
+    resumeLiveTailing,
+    clearLogs
+  } = useLiveTailingLogs(query, rowsPerPage);
+
+  const displayFieldsString = searchParams.get(LOGS_URL_PARAMS.DISPLAY_FIELDS) || LOGS_DISPLAY_FIELDS;
+  const displayFields = useMemo(() => displayFieldsString.split(","), [displayFieldsString]);
+
+  const handleResumeLiveTailing = useCallback(() => {
+    throttledScrollToBottom();
+    resumeLiveTailing();
+  }, [resumeLiveTailing]);
+
+  const handleSetRowsPerPage = useCallback((limit: number) => {
+    setSearchParamsFromKeys({ rows_per_page: limit });
+  }, [setRowsPerPage, setSearchParamsFromKeys]);
+
+  const handleSetCompactTailing = useCallback((value: boolean) => {
+    setSearchParamsFromKeys({ compact_tailing: Number(value) });
+  }, [setSearchParamsFromKeys]);
+
+  useEffect(() => {
+    startLiveTailing();
+    return () => stopLiveTailing();
+  }, [startLiveTailing, stopLiveTailing]);
+
+  useEffect(() => {
+    const container = containerRef.current;
+    if (!container) return;
+
+    const handleScroll = () => {
+      const { scrollTop, scrollHeight, clientHeight } = document.documentElement;
+      const isBottom = Math.abs(scrollHeight - scrollTop - clientHeight) < SCROLL_THRESHOLD;
+
+      setIsAtBottom(isBottom);
+
+      if (!isBottom && !isPaused) {
+        pauseLiveTailing();
+      }
+    };
+
+    document.addEventListener("scroll", handleScroll);
+    return () => document.removeEventListener("scroll", handleScroll);
+  }, [isPaused, pauseLiveTailing, resumeLiveTailing]);
+
+  useEffect(() => {
+    if (isAtBottom && !isPaused) {
+      throttledScrollToBottom();
+    }
+  }, [logs, isAtBottom]);
+
+  useEffect(() => {
+    handleResumeLiveTailing();
+  }, [rowsPerPage]);
+
+
+  const renderSettings = () => {
```

Comment:
What do you think about moving the settings into a separate component?

Reply:
I thought about this when I created this component, but decided to leave it because almost all of the component's states need to be passed as props.

But will try to do it, I think it will be better then keep all code in one component.